packages/smooth_app/lib/database/dao_transient_folksonomy.dart

```diff
@@ -0,0 +1,80 @@
+import 'dart:convert';
+
+import 'package:hive/hive.dart';
+import 'package:openfoodfacts/openfoodfacts.dart';
+import 'package:smooth_app/database/abstract_dao.dart';
+import 'package:smooth_app/pages/folksonomy/folksonomy_provider.dart';
+
+class FolksonomyOperation {
+  FolksonomyOperation({required this.type, required this.tag});
+
+  final FolksonomyAction type;
+  final ProductTag tag;
+
+  Map<String, dynamic> toJson() => <String, dynamic>{
+    'type': type.name,
+    'tag': tag.toJson(),
+  };
+
+  static FolksonomyOperation fromJson(Map<String, dynamic> json) =>
+      FolksonomyOperation(
+        type: FolksonomyAction.values.byName(json['type'] as String),
+        tag: ProductTag.fromJson(json['tag'] as Map<String, dynamic>),
+      );
+}
+
+class TransientFolksonomy {
+  const TransientFolksonomy(this.barcode, this.operations);
+
+  final String barcode;
+  final String operations;
+}
+
+class DaoTransientFolksonomy extends AbstractDao {
+  DaoTransientFolksonomy(super.localDatabase);
+
+  static const String _hiveBoxName = 'transientFolksonomyOperations';
+
+  @override
+  Future<void> init() async => Hive.openBox<String>(_hiveBoxName);
+
+  @override
+  void registerAdapter() {
+    // We don't need it here since we encode/decode Strings.
+  }
+
+  Box<String> _getBox() => Hive.box<String>(_hiveBoxName);
+
+  List<FolksonomyOperation> get(final String barcode) =>
+      _getFolksonomyOperationsFromJson(_getBox().get(barcode) ?? '[]');
```

Comment:
```suggestion
  List<FolksonomyOperation>? get(final String barcode) {
    final String? value = _getBox().get(barcode);
    if (value == null) {
      return null;
    }
    return _getFolksonomyOperationsFromJson(value);
  }
```

Looks like we need to know the difference between empty and null later in the code. If it's empty, we can delete it. If it's null, we don't even have to.